controller/api/destination/watcher/workload_watcher.go

```diff
@@ -310,6 +298,9 @@ func (ww *WorkloadWatcher) updateServer(oldObj interface{}, newObj interface{})
 
 	oldUpdated := latestUpdated(oldServer.ManagedFields)
 	updated := latestUpdated(newServer.ManagedFields)
+	if oldServer.ResourceVersion == newServer.ResourceVersion {
+		return
+	}
```

Comment:
As far as I understand, you are already minimizing the amount of work dispatched by checking whether there is a change in the "opaqueness" in `updateServers`. Why go further and ignore resyncs too? If I remember correctly, we discussed resync ignores when it came to the endpoints slices controller, and reached the conclusion that it is fine to skip them if we have another retry mechanism (a retry queue), which we do not have here. Here are some follow up questions: 

1. If we are skipping resyncs here, why not skip them for all update callbacks (e.g. pods)
2. Do we have any idea how much of the  work needlessly ends up in the update queue due to the fact that we were not checking opaqueness state change as opposed to not ignoring resyncs?. In other words, if you let resyncs go through, would the rest of the changes still solve the problem that we were seeing?